ext/semian/semian.c

```diff
@@ -65,7 +67,19 @@ void Init_semian()
   /* Maximum number of tickets available on this system. */
   rb_define_const(cSemian, "MAX_TICKETS", INT2FIX(system_max_semaphore_count));
 
-  Init_SimpleInteger();
-  Init_SlidingWindow();
-  Init_CircuitBreaker();
+  if (use_c_circuits()) {
+    Init_SimpleInteger();
+    Init_SlidingWindow();
+    Init_CircuitBreaker();
+  }
+}
+
+static int
+use_c_circuits() {
+  char *circuit_impl = getenv("CIRCUIT_IMPL");
```

Comment:
Nit: Since Semian is a library, running in the context of a larger program, I think it makes sense to namespace this variable. Perhaps something like `SEMIAN_CIRCUIT_BREAKER_IMPL`?